comp/core/statsd/component.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package statsd implements a component to get a statsd client.
+package statsd
+
+import (
+	ddgostatsd "github.com/DataDog/datadog-go/v5/statsd"
+)
+
+// team: agent-shared-components
+
+// Component is the component type.
+type Component interface {
```

Comment:
I was going to make the same comment, but then I saw this in @iksaif conversation with @gh123man above:
> We should not create an instance of the statsd client into the component because there are valid reasons to want different client with different settings.

Can you elaborate a little bit on this requirement in the scope of a certain process?

Reply:
A good example is having different clients for the part of your code that sends a lot of metrics and the part that sends vital metrics (since agent fairness will try not to drop the client with little traffic - it's already the case with unix streams)

Another is an application using ChannelMode everywhere, but MutexMode in very specific parts. We have those already